### ISSUE TITLE
Facebook omniauth redirection is missing url when testing on localhost

### ISSUE THREAD

**Issue by c0mrade** [lynndylanhurley/devise_token_auth#285]
Relating my previous issues :
https://github.com/lynndylanhurley/devise_token_auth/issues/166

The second part has been resolved regarding `api_user` and `user`. However first part still remains that I get redirected to a page without a hostname, just has a path.

Solution to this problem could be in the `OmniauthCallbacksController`, method `redirect_callbacks`, this line :
```
redirect_route = "/#{Devise.mappings[devise_mapping].as_json["path"]}/#{params[:provider]}/callback"
```

change it to :
```
redirect_route = "#{request.protocol}#{request.host_with_port}/#{Devise.mappings[devise_mapping].as_json["path"]}/#{params[:provider]}/callback"
```

Tried it locally works, any objections, different ideas how to do this?

**Comment by c0mrade** [comment]
@lynndylanhurley can you see any issues with adding request protocol and port to the redirect_route?

**Comment by booleanbetrayal** [comment]
@c0mrade - I don't see any issues, and it should probably account for the full URL info anyhow. Do you mind putting together a quick PR and making sure all existing tests pass?

**Comment by c0mrade** [comment]
@booleanbetrayal sure, thanks for the response

**Comment by booleanbetrayal** [comment]
Should be closed as of #322
